pkg/stanza/fileconsumer/internal/reader/reader.go

```diff
@@ -53,6 +53,7 @@ type Reader struct {
 	includeFileRecordNum   bool
 	compression            string
 	acquireFSLock          bool
+	new                    bool // indicates that a reader is new (no previously known offset)
```

Comment:
@djaglowski there is an alternative implementation as well. We can update[newReader](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e6c14a596e56b4489d8fa6e08ff7b29c4fd1f07a/pkg/stanza/fileconsumer/file.go#L235-L272) implementation to return a boolean indicating new or matched reader. We can then add it to a separate `fileset`, which'll track only unmatched files.